src/o365/spo/SpoCommand.ts

```diff
@@ -124,4 +127,110 @@ export default abstract class SpoCommand extends Command {
         });
     });
   }
+
+  public ensureFormDigest(cmd: CommandInstance, context: FormDigestInfo, debug: boolean): Promise<FormDigestInfo> {
+    return new Promise<FormDigestInfo>((reject: (error: any) => void): void => {
+      if (this.isUnexpiredFormDigest(context)) {
+
+        if (debug) {
+          cmd.log('Existing form digest still valid');
+        }
+
+        Promise.resolve(context);
+      }
+
+      this
+        .getRequestDigest(cmd, debug)
+        .then((res: FormDigestInfo): void => {
+          if (debug) {
+            cmd.log('Response:');
+            cmd.log(res);
+            cmd.log('');
+          }
+
+          context.FormDigestValue = res.FormDigestValue;
+          context.FormDigestTimeoutSeconds = res.FormDigestTimeoutSeconds
+          context.FormDigestExpiresAt = new Date();
+          context.FormDigestExpiresAt.setSeconds(context.FormDigestExpiresAt.getSeconds() + res.FormDigestTimeoutSeconds - 5);
+
+          Promise.resolve(context);
+        }, (error: any): void => {
+          reject(error);
+        });
+    });
+  }
+
+  private isUnexpiredFormDigest(contextinfo: FormDigestInfo): boolean {
+    const now: Date = new Date();
+    if (contextinfo.FormDigestValue &&
+      now < (contextinfo.FormDigestExpiresAt as Date)) {
```

Comment:
Why is it necessary to cast `FormDigestExpiresAt` as Date when it's already defined as Date?

Reply:
My bad, lazy 'copy paste'